### DISCUSSION TITLE
Adding recaptcha to custom form

### DISCUSSION THREAD

**Discussion by mpurish** [breezefront/community#341] [1. Help]
Hi,

I've built a custom contact form that is submitted with ajax. How can I add built-in captcha from Magento to it? Basically it's a small module with a template, layout and controller that sends an email.

Thanks!

**Comment by vovayatsyuk** [comment]
Hello,

It's not a question about Breeze. You need a working functionality on Luma, then you can integrate it with Breeze.

Here is a StackOverflow answer on how to implement recaptcha in custom form: https://magento.stackexchange.com/questions/133238/magento-2-how-to-add-captcha-to-a-custom-form/340489#340489

Since this answer doesn't use js code directly, there is a big chance that this solution with work in Breeze too.

**Reply by mpurish** [reply]
Ok, thank you, I'll take a look into that

**Reply by mpurish** [reply]
Hello again @vovayatsyuk, I tried suggested solution and it works in Luma, but when I switch to Breeze captcha block is not being rendered on frontend, seems like knockout.js template is not loaded. First 2 screenshots attached are from Breeze, the last 2 are from Luma.


<img width="1356" alt="Screenshot 2024-12-02 at 21 02 21" src="https://github.com/user-attachments/assets/e7bb6d95-f1c2-4424-8432-a3c4a6d36686">
<img width="441" alt="Screenshot 2024-12-02 at 21 02 35" src="https://github.com/user-attachments/assets/37af3ee9-cca4-49ce-819c-97fabefeed54">

<img width="827" alt="Screenshot 2024-12-02 at 21 00 11" src="https://github.com/user-attachments/assets/c935de3f-c7f8-45f5-9d02-585b1abf0afe">
<img width="381" alt="Screenshot 2024-12-02 at 21 00 23" src="https://github.com/user-attachments/assets/e9b74719-9199-4dac-998f-ac454438cb86">

**Reply by mpurish** [reply]
Please never mind, I tweaked .xml layout a bit and the ko template rendered without issues. Thank you
